src/lib/sfge/AbstractSfgeEngine.ts

```diff
@@ -215,19 +215,47 @@ export abstract class AbstractSfgeEngine extends AbstractRuleEngine {
 		try {
 			// Execute graph engine
 			const output = await SfgeExecuteWrapper.runSfge(targets, filteredRules, sfgeConfig);
-			results = this.processStdout(output);
+			// If the transaction succeeded, then just use whatever results we find
+			// inside the output.
+			results = this.parseViolations(output);
 		} catch (e) {
-			// Handle errors thrown
-			const message = e instanceof Error ? e.message : e as string;
-			this.logger.trace(`${this.getSubVariantName()} evaluation failed. ${message}`);
-			await this.eventCreator.createUxErrorMessage('error.external.sfgeIncompleteAnalysis', [AbstractSfgeEngine.processStderr(message)]);
-			// Handle output results no matter the outcome.
-			results = this.processStdout(message);
+			// Get the message from the error.
+			const message: string = e instanceof Error ? e.message : e as string;
+			// The error message contains both stdout and stderr, so we have to process
+			// it as both of those things.
+			results = await this.processExecutionFailure(message);
 		}
 		this.logger.trace(`Found ${results.length} results for ${this.getSubVariantName()}`);
 		return results;
 	}
 
+	/**
+	 * Accepts the complete output of a failed transaction.
+	 * If partial results can be found in the output, they are returned and any errors
+	 * are logged to the console.
+	 * If no partial results are found, then the error is rethrown.
+	 * @param rawErrorMessage Per {@link SfgeExecuteWrapper.handleResults}, the concatenated {@code stdout} and {@code stderr} from a failed execution.
+	 * @protected
+	 */
+	protected async processExecutionFailure(rawErrorMessage: string): Promise<RuleResult[]> {
+		this.logger.trace(`${this.getSubVariantName()} evaluation failed. ${rawErrorMessage}`);
+		// Try to pull partial results from the message.
+		const results: RuleResult[] = this.parseViolations(rawErrorMessage);
+		// Pull the error from the message.
+		const errorMessage = AbstractSfgeEngine.parseError(rawErrorMessage);
+
+		// If there are no results, then we assume the error was fatal and rethrow it,
+		// to forcibly bring it to the user's attention.
+		if (results.length === 0) {
+			throw new Error(errorMessage);
+		}
```

Comment:
This is the net-new behavior. Previously, if there were no partial results, we'd return an empty array and just log the errors.